### PR TITLE
Bump client-go and library-go to current 4.6 tips

### PR DESCRIPTION
**Pull request by wking**
Builds on #405; you may want to review that first.

Pulling in openshift/client-go@83993cebb5aee53 and openshift/library-go@cc498c355c9998.  Generated with:

```context
$ go get -u github.com/openshift/client-go@83993cebb5aee533bf2f2dded2d87b9e32835f4a
go: github.com/openshift/client-go 83993cebb5aee533bf2f2dded2d87b9e32835f4a => v0.0.0-20200623090625-83993cebb5ae
$ go get -u github.com/openshift/library-go@cc498c355c99983057e9e01f3abcceb40ca5c298
go: github.com/openshift/library-go cc498c355c99983057e9e01f3abcceb40ca5c298 => v0.0.0-20200709151624-cc498c355c99
$ go mod vendor
go: downloading k8s.io/apimachinery v0.18.4
go: downloading github.com/openshift/api v0.0.0-20200623075207-eb651a5bb0ad
go: downloading k8s.io/api v0.18.4
go: downloading golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading golang.org/x/text v0.3.3
$ go mod tidy
$ git add -A go.* vendor
```

using:

```context
$ go version
go version go1.14.4 linux/amd64
```

The client-go bump in turn addresses the golang.org/x/text v0.3.3 to address [CVE-2020-14040][3].

```context
$ grep '# golang.org/x/text' vendor/modules.txt
```

WIP is because I still need things like `Context` arguments to keep up with the Kube-client API, which has been pulled from v0.17.1 to v0.18.3.  Will get to that in a follow-up commit, and then pull the WIP from this PR.

[1]: https://github.com/openshift/client-go/commit/83993cebb5aee533bf2f2dded2d87b9e32835f4a
[2]: https://github.com/openshift/library-go/commit/cc498c355c99983057e9e01f3abcceb40ca5c298
[3]: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-14040

### CODE REVIEW

**Comment by wking** [comment]
Reopened and rebased on the master to update our dependencies and remove the #408 pinning.  Diff is huge, but the first commit is the automated vendor bump, and the second is me massaging *so many* call signatures to add `Context` and `*Options{}` to keep up.  There are a handful of `context.TODO()` where I have to wrap a modern function to fit into a legacy lister interface; I imagine we'll be able to drop those in some future vendor bump.

Because it's sort of in the spirit of #410:

/assign @jottofar

File: pkg/start/start.go [hunk 1]
```diff
@@ -229,9 +229,9 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 					time.Sleep(100 * time.Millisecond)
 					// if we still hold the leader lease, clear the owner identity (other lease watchers
 					// still have to wait for expiration) like the new ReleaseOnCancel code will do.
-					if err := lock.Update(resourcelock.LeaderElectionRecord{}); err == nil {
+					if err := lock.Update(localCtx, resourcelock.LeaderElectionRecord{}); err == nil {
 						// if we successfully clear the owner identity, we can safely delete the record
-						if err := lock.Client.ConfigMaps(lock.ConfigMapMeta.Namespace).Delete(lock.ConfigMapMeta.Name, nil); err != nil {
+						if err := lock.Client.ConfigMaps(lock.ConfigMapMeta.Namespace).Delete(ctx, lock.ConfigMapMeta.Name, metav1.DeleteOptions{}); err != nil {
```
Comment (jottofar) [inline pkg/start/start.go:234]: I understand the use of `localCtx` but why then the use of the parent `ctx`?  Assuming there's a good reason, and given there are so many Context variables in scope, a comment would be helpful.

Comment (wking) [reply pkg/start/start.go]: > I understand the use of `localCtx` but why then the use of the parent `ctx`?

Fixed to be `localCtx` with 23ba2bda -> 27b0531f.

**Comment by wking** [comment]
Last three e2e failures are split between `oc adm must-gather runs successfully for audit logs` [here][1], `DisruptionController should block an eviction until the PDB is updated to allow it` [here][2], and both and some more besides [here][3].  The former is [rhbz#1861201][4], the later is [rhbz#1861189][5].  Neither is related to this change.

/override ci/prow/e2e

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/406/pull-ci-openshift-cluster-version-operator-master-e2e/1287918162578247680
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/406/pull-ci-openshift-cluster-version-operator-master-e2e/1287865991518228480
[3]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/406/pull-ci-openshift-cluster-version-operator-master-e2e/1287892362810363904
[4]: https://bugzilla.redhat.com/show_bug.cgi?id=1861201
[5]: https://bugzilla.redhat.com/show_bug.cgi?id=1861189

**Pull request merged**